orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/HttpClientUtils.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.util;
+
+import com.google.common.io.CharStreams;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.ServiceUnavailableRetryStrategy;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.protocol.HttpCoreContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.Arrays;
+import java.util.List;
+
+
+public class HttpClientUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientUtils.class);
+  private static final int MAX_RETRIES = 5;
+  private static final int RETRY_INTERVAL = 3000;
+  private static final int TIMEOUT_MILLIS = 3000;
+  private static List<Integer> RETRYABLE_500_HTTP_STATUS_CODES =Arrays.asList(
+      HttpStatus.SC_SERVICE_UNAVAILABLE,
+      HttpStatus.SC_INTERNAL_SERVER_ERROR
+  );
+
+  private static CloseableHttpClient httpClient = httpClientWithServiceUnavailableRetryStrategy();
+
+  private static CloseableHttpClient httpClientWithServiceUnavailableRetryStrategy() {
+    HttpClientBuilder httpClientBuilder =  HttpClients.custom()
+      .setServiceUnavailableRetryStrategy(new ServiceUnavailableRetryStrategy() {
```

Comment:
Is there a retry strategy for IOExceptions we could also add (where we didn't get an HTTP response back to determine the code)